### ISSUE TITLE
App breaks when resizing the window with an aspect ratio

### ISSUE THREAD

**Issue by Sandakan** [electron/electron#38098]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

24.1.2

### What operating system are you using?

Windows

### Operating System Version

Windows 10 Pro 10.0.19045 Build 19045

### What arch are you using?

x64

### Expected Behavior

The app comes to the previous window size while following the set aspect ratio.

### Actual Behavior

The app breaks with an exit code 3221226525

### Testcase Gist URL

https://gist.github.com/Sandakan/387f88a146c778fda6d29abd8c6eb4a9

### Additional Information

https://user-images.githubusercontent.com/45460443/233906729-87d92a98-65b9-4b7c-966b-50f5287f97d8.mov

**Comment by codebytere** [comment]
@Sandakan does this happen on other versions? i've tried with the provided version on Windows and cannot reproduce the bug so far. The window resizes and snaps to the preset aspect ratio without issue.

**Comment by haroldiedema** [comment]
I experienced something similar that may be related to this. When listening to the `resize`-events (will or did) of the `BrowserWindow`, these events may/may not be triggered properly when the window is being "snapped" as seen in the video. This happens on Windows 11 (latest updates) and on the latest stable electron version.

I noticed this since I persist the window size after a resize to a configuration file and noticed that the last window size wasn't set properly when a window is being snapped. The strange thing is, is that this happens only 'sometimes'. E.g., from maximized to a snapped position or vice-versa.

**Comment by VerteDinde** [comment]
Hey @Sandakan, we so far haven't been able to repro this issue. While we're trying to get a repro, would you mind attaching a crash dump to help us get more information? You can collect them by adding the following snippet to your main process code, before app.whenReady:

```
const { app, crashReporter } = require('electron')

console.log(app.getPath('crashDumps'))
crashReporter.start({ submitURL: '', uploadToServer: false })
```
Then reproduce the crash, zip up the crash dumps directory and attach it here.

**Comment by Sandakan** [comment]
@VerteDinde  Sure.
[Crashpad.zip](https://github.com/electron/electron/files/11338843/Crashpad.zip)

**Comment by Sandakan** [comment]
Seems like this bug is fixed in v25.1.1

**Comment by Sandakan** [comment]
Fixed in newer versions
